relayer/app/sender.go

```diff
@@ -84,23 +85,33 @@ func NewSender(
 		network:      network,
 		txMgr:        txMgr,
 		gasEstimator: newGasEstimator(network),
-		portal:       chain.PortalAddress,
 		abi:          &parsedAbi,
 		chain:        chain,
 		gasToken:     meta.NativeToken,
 		chainNames:   chainNames,
-		rpcClient:    rpcClient,
+		ethCl:        rpcClient,
 		onSubmit:     onSubmit,
 	}, nil
 }
 
-// SendTransaction sends the submission to the destination chain.
-func (s Sender) SendTransaction(ctx context.Context, sub xchain.Submission) error {
+// SendAsync sends the submission to the destination chain asynchronously.
+// It returns a channel that will receive an error if the submission fails or nil when it succeeds.
+// Nonces are however reserved synchronously, so ordering of submissions
+// is preserved.
+func (s Sender) SendAsync(ctx context.Context, sub xchain.Submission) <-chan error {
+	// Helper function to return error "synchronously".
+	returnErr := func(err error) chan error {
+		resp := make(chan error, 1)
+		resp <- err
+
```

Comment:
we should close resp channel
```suggestion
close(resp)
```

Reply:
it is buffered, and we either add `err` or `nil` so I do not think we need to close it